instance/management/commands/activity_csv.py

```diff
@@ -70,13 +70,19 @@ def handle(self, *args, **options):
 
         self.activity_csv(out)
 
-    def activity_csv(self, out):  # pylint: disable=missing-docstring
-        # Produce a mapping of public IPs (of active app servers) to parent instances.
-        active_appservers = {
-            instance.active_appserver.server.public_ip: instance for instance in OpenEdXInstance.objects.all()
-            if not (instance.active_appserver is None or instance.active_appserver.server.public_ip is None)
-        }
-
+    @staticmethod
+    def get_active_appservers():
+        """ Produce a mapping of public IPs (of active app servers) to parent instances. """
+        active_appservers = {}
+        for appserver in OpenEdXAppServer.objects.filter(_is_active=True):
+            public_ip = appserver.server.public_ip
+            if public_ip is not None:
+                active_appservers[public_ip] = appserver.instance
+        return active_appservers
+
+    def activity_csv(self, out):
+        """Generate the activity CSV."""
+        active_appservers = self.get_active_appservers()
```

Comment:
@antoviaque I don't think an appserver set matters here in any way.  We just want to look at all the active appservers.  We don't care whether they are grouped in any way.